internal/dataplane/sendconfig/config_change_detector.go

```diff
@@ -64,30 +70,31 @@ func (d *DefaultConfigurationChangeDetector) HasConfigurationChanged(
 	if err != nil {
 		return false, fmt.Errorf("failed to verify kong readiness: %w", err)
 	}
-	// Kong instance has no configuration, we should push despite the oldSHA and newSHA being equal.
+
+	// Kong instance has no configuration, we should push despite the oldSHA and newSHA being equal...
 	if hasNoConfiguration {
+		// ... unless we're trying to push an empty config in such case skip.
+		if len(targetConfig.CACertificates) == 0 &&
+			len(targetConfig.Certificates) == 0 &&
+			len(targetConfig.ConsumerGroups) == 0 &&
+			len(targetConfig.Consumers) == 0 &&
+			len(targetConfig.Plugins) == 0 &&
+			len(targetConfig.PluginConfigs) == 0 &&
+			len(targetConfig.RBACRoles) == 0 &&
+			len(targetConfig.Routes) == 0 &&
+			len(targetConfig.ServicePackages) == 0 &&
+			len(targetConfig.Services) == 0 &&
+			len(targetConfig.Upstreams) == 0 &&
+			len(targetConfig.Vaults) == 0 {
+			return false, nil
```

Comment:
Can we make this check more future-proof? Now, if the `file.Content` will be extended with some new entities (e.g. `License` that Travis is working on), we may unintentionally skip sending the config even when we have something in there. 

Maybe `cmp.Diff` comparing against a zero-value struct and an option that would make it ignore the fields that we don't want to compare (I believe those would be `FormatVersion`, `Info` etc.)?

Reply:
Good call! Updated this check. PTAL